esmvaltool_sample_data/sample_data.py

```diff
@@ -112,6 +109,26 @@ def select_host(hosts, preferred_hosts, ignore_hosts):
     return hosts[0]
 
 
+def select_latest_versions(datasets: dict) -> dict:
+    """Return a dict with only the latest version of each dataset.
+
+    If `datasets` contains multiple versions of a dataset, return only
+    the most recent version.
+    """
+    keys = (key.rsplit('.', 1) for key in list(datasets.keys()))
```

Comment:
```suggestion
    keys = (key.rsplit('.', 1) for key in datasets)
```
iterating over a dict will always iterate over the keys, I believe there may be no need to specifically request the keys and convert them to a list first

Reply:
Yeah, good point. I keep forgetting about that!